topi/python/topi/mali/conv2d.py

```diff
@@ -138,20 +138,15 @@ def _schedule_spatial_pack(cfg, s, output, conv, data_vec, kernel_vec):
         s[data_vec].unroll(vw)
 
     if isinstance(kernel_vec.op, tvm.te.ComputeOp) and kernel_vec.name == 'kernel_vec':
-        if autotvm.GLOBAL_SCOPE.in_tuning:
-            # kernel packing will be pre-computed during compilation, so we skip
-            # this part to make tuning records correct
-            s[kernel_vec].pragma(s[kernel_vec].op.axis[0], 'debug_skip_region')
-        else:
-            max_threads = tvm.target.Target.current(allow_none=False).max_num_threads
-            co, ci, kh, kw, vc = s[kernel_vec].op.axis
-            fused = s[kernel_vec].fuse(co, ci, kh, kw, vc)
-            fused, vec = s[kernel_vec].split(fused, VC)
-            bb, tt = s[kernel_vec].split(fused, max_threads)
-            s[kernel_vec].bind(bb, te.thread_axis("blockIdx.x"))
-            s[kernel_vec].bind(tt, te.thread_axis("threadIdx.x"))
-            if VC in vec_size:
-                s[kernel_vec].vectorize(vec)
+        max_threads = tvm.target.Target.current(allow_none=False).max_num_threads
```

Comment:
Thank you for working on this. You also need to replace kernel with the correct layout placeholder in compute when autotvm.GLOBAL_SCOPE.in_tuning is True. Look at https://github.com/apache/incubator-tvm/pull/5200/files#diff-7d0b52f92ecc0d246c17fa17f90d2a8fR189

Reply:
Hi @kevinthesun 

Thanks for your kind feedback. I thought it is a schedule of spatial packing for conv2d, so that, I remove the `autotvm.GLOBAL_SCOPE.in_tuning` here. Just like: https://github.com/apache/incubator-tvm/pull/5200/files#diff-1646331f629f3c6af025167c79d09427L98

Can you elaborate a bit about what do you mean?

many thanks,

Reply:
While doing autotvm, all the workloads fetched are in the original data/kernel layouts, for example NCHW/OIHW. It means we need to do layout conversion first. For spatial_pack this is done inside compute. However, we need to skip this layout conversion stage while doing autotuning. Previously we use debug_skip_region which will cause inaccurate measurement. Another way to skip it is to change the input kernel tensor to a new placeholder with converted layout.

Reply:
Hi @kevinthesun 
Thanks for the kind explanation. 

Should we keep removing the `autotvm.GLOBAL_SCOPE.in_tuning:` block here, and replace kernel with the correct layout placeholder in below position?


https://github.com/apache/incubator-tvm/blob/418da931c1806b35266bc7e12c3c8ffb01fa034b/topi/python/topi/mali/conv2d.py#L456

Reply:
We need to do the replacement in compute.